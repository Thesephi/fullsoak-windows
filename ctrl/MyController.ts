import { Controller, Get, ssr } from "fullsoak";
import { MyApp } from "../things/MyApp.tsx";

@Controller()
export default class MyController {
  @Get("/")
  serve() {
    return ssr(MyApp);
  }
}
