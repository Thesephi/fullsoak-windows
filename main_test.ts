import { superoak } from "superoak";
import { assertSnapshot } from "@std/testing/snapshot";
import { useFullSoak } from "fullsoak/testing";
import MyController from "./ctrl/MyController.ts";

Deno.test(async function MyApp(t) {
  const controllers = [MyController];
  const app = await useFullSoak({ controllers });
  const req = await superoak(app);
  const res = await req
    .get("/")
    .expect(200);
  await assertSnapshot(t, res.text);
});
