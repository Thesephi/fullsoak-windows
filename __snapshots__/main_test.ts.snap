export const snapshot = {};

snapshot[`MyApp 1`] = `
'<!doctype html><html><head><meta charset="utf-8"/><meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0"/><script type="importmap">{
    "imports": {
      "preact": "https://esm.sh/preact@10.26.4",
      "preact/hooks": "https://esm.sh/preact@10.26.4/hooks",
      "preact/debug": "https://esm.sh/preact@10.26.4/debug",
      "preact/devtools": "https://esm.sh/preact@10.26.4/devtools",
      "react": "https://esm.sh/preact@10.26.4",
      "react/jsx-runtime": "https://esm.sh/preact@10.26.4/jsx-runtime",
      "preact/jsx-runtime": "https://esm.sh/preact@10.26.4/jsx-runtime",
      "react-dom": "https://esm.sh/preact@10.26.4/compat/",
      "react-dom/*": "https://esm.sh/preact@10.26.4/compat/*",
      "htm/preact": "https://esm.sh/htm@3.1.1/preact?external=preact",
      "preact-iso": "https://esm.sh/preact-iso@^2.9.0?external=preact",
      "preact-iso/prerender": "/fullsoak",
      "@preact/signals": "https://esm.sh/@preact/signals",
      "fullsoak/preact-iso": "https://esm.sh/preact-iso@^2.9.0?external=preact",
      "@fullsoak/fullsoak/preact-iso": "https://esm.sh/preact-iso@^2.9.0?external=preact",
      "fullsoak": "/fullsoak",
      "@fullsoak/fullsoak": "/fullsoak"
    }
  }</script><link rel="modulepreload" href="/components/MyApp/mount" as="script" type="text/javascript"/><link rel="modulepreload" href="/components/MyApp/index.tsx" as="script" type="text/javascript"/><script type="text/javascript"></script><style></style></head><body><main id="fullsoak-htmlshell-main"><div>My App</div><div class="box">Box 0</div><button type="button">up</button></main><script type="text/javascript">window.preloadedProps = {}</script><script defer type="module" src="/components/MyApp/mount"></script></body></html>'
`;
